src/mongosh/schemas/curriculum-2.0.0.json

```diff
@@ -0,0 +1,103 @@
+{
+	"description": "A Learning resource or other activity that is on a person's career or learning roadmap",
+	"bsonType": "object",
+	"required": [
+		"_id",
+		"resources"
+	],
+	"properties": {
+		"_id": {
+			"description": "The unique identifier for a resource used by a person",
+			"bsonType": "objectId"
+		},
+		"name": {
+			"description": "Temporary property for future removal",
+			"bsonType": "string"
+		},
+		"version": {
+			"description": "Schema Version (only on Version document)",
+			"bsonType": "string",
+			"maxLength": 20
+		},
+		"resources": {
+			"description": "An array of resources",
+			"bsonType": "array",
+			"items": {
+				"bsonType": "object",
+				"required": ["type", "review", "roadmap"],
+				"properties": {
+					"type": {
+						"description": "Type of the resource (topic or adhoc)",
```

Comment:
I think this is your first problem, type does not have a "bsonType" property which makes this schema invalid.